File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        
        Set<Integer> set1 = new HashSet<>();
        for(Integer val : nums1){
            set1.add(val);
        }
        
        Set<Integer> set2 = new HashSet<>();
        for(Integer val : nums2){
            set2.add(val);
        }
        
        Set<Integer> finalSet = new HashSet<>(set1);
        finalSet.retainAll(set2);
        
        int[] result = new int[finalSet.size()];
        int index = 0;
        for(Integer val : finalSet){
            result[index] = val;
            index++;
        }
        
        return result;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cs
public class Solution {
    public int[] Intersection(int[] nums1, int[] nums2) {
        HashSet<int> set1 = new HashSet<int>();
        foreach(int val in nums1){
            set1.Add(val);
        }
        
        HashSet<int> set2 = new HashSet<int>();
        foreach(int val in nums2){
            set2.Add(val);
        }
        
        HashSet<int> finalSet = new HashSet<int>(set1);
        finalSet.IntersectWith(set2);
        
        int[] result = new int[finalSet.Count];
        int index = 0;
        foreach(int val in finalSet){
            result[index] = val;
            index++;
        }
        
        return result;
    }
}